app/javascript/controllers/audio_breakpoints_controller.js

```diff
@@ -6,39 +6,61 @@ export default class extends Controller {
 
   static values = {
     labelPrefix: { type: String, default: "Breakpoint" },
-    segmentCount: { type: Number, default: 1 },
+    adBreaks: { type: Number, default: 1 },
     markers: Array,
   }
 
+  connect() {
+    this.initMarkers()
+  }
+
   /**
    * Handle external changes to incoming marker values.
    * Should only run when controller initializes, or we need to edit ad locations for a different episode.
    * DO NOT UPDATE THIS VALUE AS A RESULT OF EDITS IN THE AD LOCATIONS UI.
    */
   markersValueChanged() {
+    if (!this.breakpointMarkers) return
     this.initMarkers()
   }
 
-  segmentCountValueChanged() {
+  adBreaksValueChanged() {
+    if (!this.breakpointMarkers) return
     this.initMarkers()
   }
 
   /**
    * Initial render of ad markers.
    */
   initMarkers() {
-    this.breakpointMarkers = [...Array(this.segmentCountValue - 1).keys()]
-      .map((key) => this.markersValue?.[key] || [])
+    if (!this.hasMarkersValue || !this.hasAdBreaksValue) return
```

Comment:
May want to also do some light validation on this?  I'm noting I can (by typing, not clicking up/down) input `-3` etc.  And if i just add a few zeroes to it, the UI lags quite awhile trying to render 100s of these.

Or alternatively, pass in a value that is `ad-breaks-valid-value="<%= episode.validate(:ad_breaks) %>"`.

Reply:
Did know there was a validate method on the model. Added that to the existing controller value attribute.

Reply:
Added `adBreaksValid` value and use it to prevent initializing/updating UI with invalid number of ad breaks.